subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationWorker.java

```diff
@@ -16,16 +16,18 @@
 
 package org.gradle.internal.operations;
 
-import org.gradle.api.Action;
+import org.gradle.api.Nullable;
 
 /**
- * A worker that can run build operations of type {@link T}.
+ * A worker that can run build operations.
  *
  * Implementations must be thread-safe.
  */
-public interface BuildOperationWorker<T extends BuildOperation> extends Action<T> {
+public interface BuildOperationWorker<O extends BuildOperation> {
     /**
      * Returns a human consumable name for this tool.
      */
     String getDisplayName();
+
+    void execute(O buildOperation, @Nullable BuildOperationState parent);
```

Comment:
Right. When I pulled that up I did not have the full overview of how the parent/child relationship works.

I thought other worker implementations could also make use of the parent if they somehow would reconnect to the BuildOperationExecutor to create more build operations on their own. But since we do not do something like that (yet). Or might do it differently in the future, we may as well leave it like it was.

I reverted the interface change and reintroduced the ParentBuildOperationAwareWorker.